crates/ty/src/lib.rs

```diff
@@ -450,12 +450,12 @@ impl ty_project::ProgressReporter for IndicatifReporter {
         self.bar.set_draw_target(self.printer.progress_target());
     }
 
-    fn report_checked_file(&self, db: &dyn Db, file: File, diagnostics: &[Diagnostic]) {
+    fn report_checked_file(&self, db: &ProjectDatabase, file: File, diagnostics: &[Diagnostic]) {
```

Comment:
I had to change the signature of `ProgressReporter` to take a `ProjectDatabase` as argument because we now need a `ty_server::Db` instead of a `ty_project::Db` in the LSP.